quint/src/types/constraintGenerator.ts

```diff
@@ -38,14 +38,39 @@ import { getSignatures } from './builtinSignatures'
 import { Constraint, Signature, TypeScheme, toScheme } from './base'
 import { Substitutions, applySubstitution, compose } from './substitutions'
 import { LookupTable } from '../names/base'
-import { specialConstraints } from './specialConstraints'
+import {
+  fieldConstraints,
+  fieldNamesConstraints,
+  itemConstraints,
+  recordConstructorConstraints,
+  tupleConstructorConstraints,
+  withConstraints,
+} from './specialConstraints'
 import { FreshVarGenerator } from '../FreshVarGenerator'
 
 export type SolvingFunctionType = (
   _table: LookupTable,
   _constraint: Constraint
 ) => Either<Map<bigint, ErrorTree>, Substitutions>
 
+function validateArity(
+  operator: string,
```

Comment:
I think this would benefit from a brief explanation.
Also, maybe rename the parameter, to make clear it's a string.

```suggestion
// Check if an application of operator `opcode` to arguments `args`
// has correct arity. In particular, this returns `right(null)` if 
// `pred(args.length)` holds, and a `left(ErrorTree)` otherwise.
//
// Use this for operators that cannot be typed in the Quint type.
//
// `msg` contains a textual description of the expected argument
// length ("1", "2", "even number of", ...).
function validateArity(
  opcode: string,
```

Reply:
Nice comment. Thanks :)